### ISSUE TITLE
Custom URL link in Rich Text (blocks)

### ISSUE THREAD

**Issue by MrAndriy** [strapi/strapi#18562]
## Bug report

Error validation when attach link like mailto:

### Required System information

- Node.js version:  v18.12.1
- NPM version: Yarn Version: yarn/1.22.19
- Strapi version: 4.14.4
- Database: sqlite
- Operating system: darwin-arm64
- Is your project Javascript or Typescript: Javascript

### Describe the bug

Get an error when I want to pin a link like mailto:info@test.com in Rich Text (blocks) component

```json
{
    "data": null,
    "error": {
        "status": 400,
        "name": "ValidationError",
        "message": "content[1].children[1].url must be a valid URL",
        "details": {
            "errors": [
                {
                    "path": [
                        "content",
                        "1",
                        "children",
                        "1",
                        "url"
                    ],
                    "message": "content[1].children[1].url must be a valid URL",
                    "name": "ValidationError"
                }
            ]
        }
    }
}
```

### Steps to reproduce the behavior

1. Go to Rich Text (blocks)
2. Type text
3. Attach link with 

### Expected behavior

Possibility to attach tags like mailto:

**Comment by lalitkumawat1m** [comment]
Hello Team I want to work on this issue please assign to me

**Comment by Ben-Zahler** [comment]
Also, relative links (e.g. /strapi ) are not validating for two reasons:
- in the BlocksInputComponent, https:// gets added to all links that do not have http or https protocol
- in blocks-validator.ts, a very strict link validation is used.
I have proposed a fix in PR https://github.com/strapi/strapi/pull/19064, let me know what you think of this.

**Comment by gorechavka** [comment]
Hello! I am encountering a similar issue while attempting to add a callable phone link (like this `tel:+4733378901`). 
Are there any updates on this matter?

**Comment by Ben-Zahler** [comment]
hi @gorechavka ,
which strapi version are you using? I fixed this issue in 4.20.1 and I have validated this for tel - URLs such as yours.

**Comment by gorechavka** [comment]
oh, I didn't know that is has been fixed. I was using 4.20.0, gonna update it to 4.20.1 then, thanks!
